src/Symfony/Component/Mime/Tests/Encoder/PunycodeAddressEncoderTest.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mime\Tests\Encoder;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Mime\Encoder\PunycodeAddressEncoder;
+use Symfony\Component\Mime\Exception\AddressEncoderException;
+
+class PunycodeAddressEncoderTest extends TestCase
+{
+    /**
+     * @dataProvider provideData
+     */
+    public function testEncodeString(string $address, string $expected)
+    {
+        $encoder = new PunycodeAddressEncoder();
+
+        $this->assertSame($expected, $encoder->encodeString($address));
+    }
+
+    public function provideData()
```

Comment:
```suggestion
    /**
     * @return \Generator<array{0: string, 1: string}>
     */
    public function provideData(): \Generator
```

Reply:
i don't think symfony uses psalm object-like arrays/shapes notations anywhere, WDYT @nicolas-grekas ?

Reply:
nope, that'd be the 1st time, not sure that's the PR to do it

Reply:
i would say let's leave it as is for now, and think if Symfony should adopt this syntax everywhere or not. IMO, its a good think since it helps understand what a function returns exactly, rather than just "array" ( `array<arraykey, mixed>` ) 

see : https://psalm.dev/docs/annotating_code/type_syntax/array_types/#object-like-arrays
see : https://docs.hhvm.com/hack/built-in-types/shapes

Reply:
OK 👍